<launch>
  <arg name="device"             default="192.168.11.223"/>
  <arg name="ag_n"               default="0"/>
  <arg name="dataset_path"       default="/project/ws_dev/src/hl2ss/hl2ss_ros/dataset"/>
  <arg name="write_data_to_file" default="False"/>

  <!-- Stream video -->
  <node name="hl_vlc_camera_stream_ag$(arg ag_n)" pkg="hl2ss_ros" type="client_stream_rm_vlc_multi.py" output="screen">
    <param name="host" value="$(arg device)"/>
    <param name="ag_n" value="$(arg ag_n)"/>
    <param name="dataset_path" value="$(arg dataset_path)"/>
    <param name="write_data_to_file" value="$(arg write_data_to_file)"/>

    <!-- Stream Operating Mode
    0: Device data (e.g. video)
    1: Device data + location data (e.g. video + camera pose)
    2: Device constants (e.g. camera intrinsics)
    -->
    <param name="mode" value="1"/>
    <param name="divisor" value="1"/>
    <param name="profile" value="0"/>
    <param name="bitrate" value="None"/>
  </node>

</launch>
